sql/sys_vars.cc

```diff
@@ -741,6 +741,29 @@ static Sys_var_charptr_fscs Sys_character_sets_dir(
        READ_ONLY GLOBAL_VAR(charsets_dir), CMD_LINE(REQUIRED_ARG),
        DEFAULT(0));
 
+static bool check_engine_supports_temporary(sys_var *self, THD *thd, set_var *var)
+{
+  if (!var->value || var->value->is_null())
+    return false;
+  String str, *res;
+  res= var->value->val_str(&str);
+  LEX_CSTRING name;
+  res->get_value(&name);
+  plugin_ref plugin= ha_resolve_by_name(0, &name, true);
```

Comment:
first argument should be `thd` here, not 0

Reply:
Ack, thanks @vuvova